lib/rate-limiter.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const Bluebird = require('bluebird');
+
+exports.limit = (request, client, defaultRate) => {
```

Comment:
Any reason we need a new file for this? I feel like this could all go in the `onPostAuth` callback in `index.js`.